### ISSUE TITLE
VSCode/Theia extension development environment

### ISSUE THREAD

**Issue by yozlet** [gitpod-io/gitpod#162]
This may be wildly out of scope, but:

Right now, modifying and testing someone else's VSCode extension is sometimes painful, especially if there are a bunch of other system dependencies (e.g. other native tools that the extension integrates, such as a compiler).

At the same time, Theia is trying to bootstrap its own plugin ecosystem, staying fairly close to VSCode API compatibility.

As a VSCode user who dabbles in extension fixes occasionally, it'd be _so much easier_ for me to be able to run, develop and test some of this stuff in a container that already comes equipped for it. I realise that there's a huge difference between "it runs OK in this Theia environment" and "it runs OK in VSCode on all supported operating systems", but that's what CI services are for.

Short version:

1. Gitpod is great for removing friction in contributing to FOSS projects
2. VSCode/Theia extensions can have quite a large amount of dev setup friction
3. Theia/Gitpod would benefit from VSCode extension authors knowing about it and catering to it
4. ![_SYNERGY_](https://media.giphy.com/media/curhc38xMnMU8/giphy.gif)

**Comment by svenefftinge** [comment]
Did you see this post?
https://medium.com/gitpod/developing-native-ui-applications-in-gitpod-15af2967c24e
